compiler/src/main/java/com/squareup/anvil/compiler/codegen/RealAnvilContext.kt

```diff
@@ -1,9 +1,21 @@
 package com.squareup.anvil.compiler.codegen
 
+import com.squareup.anvil.compiler.CommandLineOptions
 import com.squareup.anvil.compiler.api.AnvilContext
+import org.jetbrains.kotlin.descriptors.ModuleDescriptor
 
 internal data class RealAnvilContext(
   override val generateFactories: Boolean,
   override val generateFactoriesOnly: Boolean,
-  override val disableComponentMerging: Boolean
+  override val disableComponentMerging: Boolean,
+  override val module: ModuleDescriptor
 ) : AnvilContext
+
+internal fun CommandLineOptions.toAnvilContext(
+  module: ModuleDescriptor
+): RealAnvilContext = RealAnvilContext(
+  generateFactories = generateFactories,
+  generateFactoriesOnly = generateFactoriesOnly,
```

Comment:
There might also be an opportunity for an enum here to simplify all the spots that know about both of these values, since in theory if `generateFactoriesOnly` is true, `generateFactories` should also be true. e.g.
```
enum CodeGenerationSetting {
  STANDARD,
  STANDARD_WITH_DAGGER_FACTORIES,
  DAGGER_FACTORIES_ONLY
}
```

Reply:
That would require changing `AnvilContext` and that's part of our stable API I'd like to avoid touching.